Source/ThirdParty/nativefiledialog/nfd_gtk.c

```diff
@@ -326,7 +332,47 @@ nfdresult_t NFD_ChooseDirectory( const nfdchar_t *prompt,
                                  const nfdchar_t *defaultPath,
                                  nfdchar_t **outPath )
 {
+    GtkWidget *dialog = NULL;
+    nfdresult_t result = NFD_ERROR;
+
+    if ( !gtk_init_check( NULL, NULL ) )
+    {
+        NFDi_SetError(INIT_FAIL_MSG);
+        return NFD_ERROR;
+    }
+    dialog = gtk_file_chooser_dialog_new( prompt,
+                                          NULL,
+                                          GTK_FILE_CHOOSER_ACTION_SELECT_FOLDER,
+                                          "_Cancel", GTK_RESPONSE_CANCEL,
+                                          "_Open", GTK_RESPONSE_ACCEPT,
+                                          NULL );
+
+    SetDefaultPath(dialog, defaultPath);
+
+    result = NFD_CANCEL;
+    if ( gtk_dialog_run( GTK_DIALOG(dialog) ) == GTK_RESPONSE_ACCEPT )
+    {
+        char *filename = NULL;
+        filename = gtk_file_chooser_get_filename( GTK_FILE_CHOOSER(dialog) );
+        {
+            size_t len = strlen(filename);
+            *outPath = NFDi_Malloc( len + 1 );
+            memcpy( *outPath, filename, len + 1 );
+            if ( !*outPath )
+            {
+                g_free( filename );
+                gtk_widget_destroy(dialog);
+                NFDi_SetError(NOPATH_MSG);
+                return NFD_ERROR;
+            }
+        }
+        g_free(filename);
+        result = NFD_OKAY;
+    }
+
+    gtk_widget_destroy(dialog);
 
-  return NFD_ERROR;
+    WaitForCleanup();
```

Comment:
I had to write this myself, the directory finder is an Atomic add-on and the win and osx had implementations. And I added some additional error reporting, so it can tell you why its failing.